### PR TITLE
Feature: conditional features

### PR DESCRIPTION
**Pull request by Zorin95670**
Add conditional features :

- `CF_createDiagram`: allow user to create a diagram in `leto-modelizer`.
- `CF_createDiagramFromTemplate`: allow user to create a diagram from template in `leto-modelizer`.
- `CF_createProject`: allow user to create a project in `leto-modelizer`.
- `CF_createProjectFromTemplate`: allow user to create a project from template in `leto-modelizer`.
- `CF_deleteDiagram`: allow user to delete a diagram in `leto-modelizer`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,13 +6,17 @@ Backend for leto-modelizer to manage libraries and access rights.
  
 ### Conditionnal features
```
Comment (justinehell) [inline README.md:7]: `Conditional` here too

File: changelog.md [hunk 2]
```diff
@@ -14,7 +14,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * Group entity definition.
 * Enable GitHub and enterprise GitHub authentication.
 * Role entity definition.
-* First conditionnal feature to create a diagram.
+* Setup conditional features:
+  * `CF_createDiagram`: allow user to create a diagram in `leto-modelizer`.
+  * `CF_createDiagramFromTemplate`: allow user to create a diagram from template in `leto-modelizer`.
+  * `CF_createProject`: allow user to create a project in `leto-modelizer`.
+  * `CF_createProjectFromTemplate`: allow user to create a project from template in `leto-modelizer`.
+  * `CF_deleteDiagram`: allow user to delete a diagram in `leto-modelizer`.
```
Comment (justinehell) [inline changelog.md:22]: allowS

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/init.js [hunk 3]
```diff
@@ -70,7 +70,14 @@ export async function setupDefaultRoles(configuration, Parse) {
   await initParse(configuration, Parse);
   console.log('init parse done');
 
-  const defaultRolesToInitialize = ['admin', 'CF_createDiagram'];
+  const defaultRolesToInitialize = [
+    'admin',
+    'CF_createDiagram',
```
Comment (k-y0u) [inline src/init.js:75]: re-order : project > diagram + normal > template + CRUD

File: src/init.js [hunk 4]
```diff
@@ -70,7 +70,14 @@ export async function setupDefaultRoles(configuration, Parse) {
...
+    'CF_createDiagramFromTemplate',
+    'CF_createProject',
+    'CF_createProjectFromTemplate',
+    'CF_deleteDiagram',
```
Comment (k-y0u) [inline src/init.js:79]: what about deleteProject ?

Comment (k-y0u) [reply src/init.js:79]: Or update ? or list/get ?

File: tests/e2e/features/Role.feature [hunk 5]
```diff
@@ -1,16 +1,18 @@
...
-  Scenario: Check CF_createDiagram role is initialized
-    When I request "/api/classes/_Role" with method "GET" with query parameter with masterKey
-      | key   | value                       |
-      | where | {"name":"CF_createDiagram"} |
+      | where | {"name":"<role>"} |
     Then I expect length of array body field "results" is 1
+    Examples:
+      | role                         |
+      | admin                        |
+      | CF_createDiagram             |
```
Comment (k-y0u) [inline tests/e2e/features/Role.feature:11]: re-order

File: README.md [hunk 6]
```diff
@@ -6,13 +6,17 @@ Backend for leto-modelizer to manage libraries and access rights.
...
-A conditionnal feature (abbreviated CF) is linked to an action that a user is allowed (or not) to perform.
+A conditional feature (abbreviated CF) is linked to an action that a user is allowed (or not) to perform.
 It translates into a specific system role that can't be deleted nor modified by administrators, so it can only be activated or deactivated for users.
 The CF are created according to this pattern: `CF_[action]`, where `action` is written in camelCase.
 
-List of conditionnal features:
+List of conditional features:
 
-- `CF_createDiagram`: check if user can create a diagram.
+- `CF_createDiagram`: allow user to create a diagram in `leto-modelizer`.
```
Comment (k-y0u) [inline README.md:15]: re-order

File: changelog.md [hunk 7]
```diff
@@ -14,7 +14,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * Group entity definition.
 * Enable GitHub and enterprise GitHub authentication.
 * Role entity definition.
-* First conditionnal feature to create a diagram.
+* Setup conditional features:
+  * `CF_createDiagram`: allow user to create a diagram in `leto-modelizer`.
```
Comment (k-y0u) [inline changelog.md:18]: re-order

File: src/init.js (hunk 4 above)
Comment (Zorin95670) [reply src/init.js:79]: not ask by PO

File: README.md (hunk 1 above)
Comment (Zorin95670) [reply README.md:7]: done

File: changelog.md (hunk 2 above)
Comment (Zorin95670) [reply changelog.md:22]: done

File: src/init.js (hunk 3 above)
Comment (Zorin95670) [reply src/init.js:75]: done

File: tests/e2e/features/Role.feature (hunk 5 above)
Comment (Zorin95670) [reply tests/e2e/features/Role.feature:11]: done

File: README.md (hunk 6 above)
Comment (Zorin95670) [reply README.md:15]: done

**Review by k-y0u — APPROVED** [review]

**Review by justinehell — APPROVED** [review]

**Pull request merged**